Emby.Naming/Video/VideoListResolver.cs

```diff
@@ -175,64 +175,71 @@ private IEnumerable<VideoInfo> GetVideosGroupedByVersion(List<VideoInfo> videos)
-                {
-                    var primary = videosMatchingFolder[0];
-                    var remainingVideos = videosMatchingFolder.Skip(1);
-                    var videoInfo = new VideoInfo
-                    {
-                        Name = folderName,
-                        Year = primary.Year,
-                        Files = primary.Files,
-                        AlternateVersions = new List<VideoFileInfo>(),
-                        Extras = primary.Extras
-                    };
-                    foreach (VideoInfo video in remainingVideos)
-                    {
-                        videoInfo.AlternateVersions.Add(video.Files.First());
-                        videoInfo.Extras.AddRange(video.Extras);
-                    }
+            return videos;
+            //foreach (var video in videos.OrderBy(i => i.Name))
+            //{
+            //    var match = list
+            //        .FirstOrDefault(i => string.Equals(i.Name, video.Name, StringComparison.OrdinalIgnoreCase));
+
+            //    if (match != null && video.Files.Count == 1 && match.Files.Count == 1)
+            //    {
+            //        match.AlternateVersions.Add(video.Files[0]);
+            //        match.Extras.AddRange(video.Extras);
+            //    }
+            //    else
+            //    {
+            //        list.Add(video);
+            //    }
+            //}
+
+            //return list;
+        }
 
-                    return new[] { videoInfo };
-                }
+        private bool HaveSameYear(List<VideoInfo> videos)
+        {
+            return videos.Select(i => i.Year ?? -1).Distinct().Count() < 2;
+        }
+
+        private bool IsEligibleForMultiVersion(string folderName, string testFilename)
+        {
+            testFilename = Path.GetFileNameWithoutExtension(testFilename);
+
+            if (string.Equals(folderName, testFilename, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
             }
 
-            return videos.GroupBy(v => new { v.Name, v.Year }).Select(group => new VideoInfo
+            if (testFilename.StartsWith(folderName, StringComparison.OrdinalIgnoreCase))
             {
-                // Because of the grouping, we can grab the information from the first movie and make it primary
-                // The remaining movie matches are 'alternate versions'
-                Name = group.First().Name,
-                Year = group.First().Year,
-                Files = group.First().Files,
-                AlternateVersions = group.Skip(1).Select(i => i.Files[0]).ToList(),
-                Extras = group.First().Extras.Concat(group.Skip(1).SelectMany(i => i.Extras)).ToList()
-            });
+                testFilename = testFilename.Substring(folderName.Length).Trim();
+                return testFilename.StartsWith("-", StringComparison.OrdinalIgnoreCase) || Regex.Replace(testFilename, @"\[([^]]*)\]", "").Trim() == string.Empty;
```

Comment:
```suggestion
                return testFilename.StartsWith("-", StringComparison.OrdinalIgnoreCase) || string.IsNullOrWhiteSpace(Regex.Replace(testFilename, @"\[([^]]*)\]", string.Empty));
```

Reply:
Won't this fail for the empty string?

Reply:
“Indicates whether a specified string is null, empty, or consists only of white-space characters.”
https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=netframework-4.7.2

TLDR: No

Reply:
That's misleading...